src/postgres/inventory/type/PgClassType.ts

```diff
@@ -80,9 +80,21 @@ class PgClassType extends PgType<PgRow> implements ObjectType<PgRow> {
     if (typeof pgValue !== 'object')
       throw new Error(`Postgres value of object type must be an object, not '${typeof pgValue}'.`)
 
-    return new Map(Array.from(this.fields).map<[string, mixed]>(([fieldName, field]) => {
-      return [fieldName, field.type.transformPgValueIntoValue(pgValue[field.pgAttribute.name])]
-    }))
+    // Parse official pg fields
+    const obj = {};
+    Array.from(this.fields).filter<[string, mixed]>(([fieldName, field]) => pgValue.hasOwnProperty(field.pgAttribute.name)).map<[string, mixed]>(([fieldName, field]) => {
+      obj[fieldName] = field.type.transformPgValueIntoValue(pgValue[field.pgAttribute.name])
+    })
+    // Pass through other fields unchanged
+    for (const key in pgValue) {
+      obj[key] = pgValue[key];
+    }
+
+    const pairs = [];
+    for (const key in obj) {
```

Comment:
At one point the V8 team was exploring supporting a JavaScript [StrongMode](https://docs.google.com/document/d/1Qk0qC4s_XNCLemj42FqfsRLp49nDQMZ1y7fwf5YjaI4/view). The important parts from StrongMode in this case are as follows:

> - An object literal creates a strong object.
> - A TypeError is thrown if a property in an object literal has the same name as an earlier one, unless one is a getter and the other a setter.
> - A TypeError is thrown if an attempt is made, either via the '.', the '[]' syntax, or destructuring, to read a property that does not exist on an object or its prototypes. Destructuring does not throw if a default value is provided.
> - A TypeError is thrown if an attempt is made, either via the '.', the '[]' syntax, to assign to a property that does not exist on an object nor has a setter on its prototype chain. The only exception are assignments to properties of 'this' in a class’s constructor definition (see Classes).

The thesis is that using a JavaScript object as a map for arbitrary key/value pairs is a common source of errors, is bad for performance and engine optimizations, and is inferior to an [ES6 `Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) in every way. Maps have easy to use getters, setters, deletes, existence checks (`has`), iteration, arbitrary keys (so not just strings, but numbers and objects as well), more accurate TypeScript types, and a performant underlying engine implementation. All of these things behave weirdly in JavaScript objects and have a number of corner cases.

So StrongMode got me to consider always using maps for the cases where I’d normally use a JavaScript object (like this case), and after trying it I just think it’s all around superior for cases like this 😊

The objects we pass around in PostGraphQL we have no clue what the keys or values will be statically, so it doesn’t make sense to use JavaScript objects and instead makes sense to use `Map` which is much better for representing dynamic key/value pairs.